db/common/src/main/java/io/helidon/db/common/AbstractDbExecute.java

```diff
@@ -0,0 +1,151 @@
+     *
+     * @param name name of the statement
+     * @return statement text
+     */
+    protected String statementText(String name) {
+        return statements.statement(name);
+    }
+
+    @Override
+    public DbStatement<?, DbRowResult<DbRow>> createNamedQuery(String statementName) {
+        return createNamedQuery(statementName, statementText(statementName));
+    }
+
+    @Override
+    public DbStatement<?, DbRowResult<DbRow>> createQuery(String statement) {
+        return createNamedQuery(generateName(DbStatementType.QUERY, statement), statement);
+    }
+
+    @Override
+    public DbStatement<?, CompletionStage<Optional<DbRow>>> createNamedGet(String statementName) {
+        return createNamedGet(statementName, statementText(statementName));
+    }
+
+    @Override
+    public DbStatement<?, CompletionStage<Optional<DbRow>>> createGet(String statement) {
+        return createNamedGet(generateName(DbStatementType.GET, statement), statement);
+    }
+
+    @Override
+    public DbStatement<?, CompletionStage<Long>> createNamedInsert(String statementName) {
+        return createNamedInsert(statementName, statementText(statementName));
+    }
+
+    @Override
+    public DbStatement<?, CompletionStage<Long>> createInsert(String statement) {
+        return createNamedInsert(generateName(DbStatementType.INSERT, statement), statement);
+    }
+
+    @Override
+    public DbStatement<?, CompletionStage<Long>> createNamedUpdate(String statementName) {
+        return createNamedUpdate(statementName, statementText(statementName));
+    }
+
+    @Override
+    public DbStatement<?, CompletionStage<Long>> createUpdate(String statement) {
+        return createNamedUpdate(generateName(DbStatementType.UPDATE, statement), statement);
+    }
+
+    @Override
+    public DbStatement<?, CompletionStage<Long>> createNamedDelete(String statementName) {
+        return createNamedDelete(statementName, statementText(statementName));
+    }
+
+    @Override
+    public DbStatement<?, CompletionStage<Long>> createDelete(String statement) {
+        return createNamedDelete(generateName(DbStatementType.DELETE, statement), statement);
+    }
+
+    @Override
+    public DbStatement<?, CompletionStage<Long>> createNamedDmlStatement(String statementName) {
+        return createNamedDmlStatement(statementName, statementText(statementName));
+    }
+
+    @Override
+    public DbStatement<?, CompletionStage<Long>> createDmlStatement(String statement) {
+        return createNamedDmlStatement(generateName(DbStatementType.DML, statement), statement);
+    }
+
+    @Override
+    public DbStatement<?, DbResult> createNamedStatement(String statementName) {
+        return createNamedStatement(statementName, statementText(statementName));
+    }
+
+    @Override
+    public DbStatement<?, DbResult> createStatement(String statement) {
+        return createNamedStatement(generateName(DbStatementType.UNKNOWN, statement), statement);
+    }
+
+    /**
```

Comment:
What's purpose of this method? SHA-256 is hashing algorithm, not a _name_ of something.
I can imagine some kind of caching of the statements and even their execution results, but this is not what's usually being done on JDBC driver level.